app/controllers/submit.js

```diff
@@ -606,55 +616,57 @@ export default Ember.Controller.extend(Analytics, BasicsValidations, NodeActions
-            // and resets displayed title to current node title. (No requests sent, front-end only.)
             Ember.get(this, 'metrics')
                 .trackEvent({
                     category: 'button',
                     action: 'click',
                     label: `Preprints - ${this.get('editMode') ? 'Edit' : 'Submit'} - Discard Upload Changes`
                 });
-            let currentFile = this.get('store').peekRecord('file', this.get('model.primaryFile.id'));
-            this.set('file', null);
-            this.set('selectedFile', currentFile);
-
-            this.set('nodeTitle', this.get('node.title'));
-            this.set('titleValid', true);
 
+            this.setProperties({
+                file: null,
+                selectedFile: this.get('store').peekRecord('file', this.get('model.primaryFile.id')),
+                nodeTitle: this.get('node.title'),
+                titleValid: true,
+            });
         },
+
+        //If user goes back and changes a section inside Upload, all fields downstream of that section need to clear.
         clearDownstreamFields(section) {
-            //If user goes back and changes a section inside Upload, all fields downstream of that section need to clear.
-            if (!this.get('nodeLocked')) { // Only clear downstream fields in Add mode!
-                switch (section) {
-                    case 'allUpload':
-                        this.set('node', null);
-                        this.set('selectedFile', null);
-                        this.set('file', null);
-                        this.set('convertOrCopy', null);
-                        this.set('nodeTitle', null);
-                        break;
-                    case 'belowNode':
-                        this.set('selectedFile', null);
-                        this.set('file', null);
-                        this.set('convertOrCopy', null);
-                        this.set('nodeTitle', null);
-                        break;
-                    case 'belowFile': {
-                        this.set('convertOrCopy', null);
-                        this.set('nodeTitle', null);
-                        break;
-                    }
-                    case 'belowConvertOrCopy': {
-                        this.set('nodeTitle', null);
-                        break;
-                    }
-                }
+            // Only clear downstream fields in Add mode!
+            if (this.get('nodeLocked'))
+                return;
+
+            const props = [];
+
+            /* eslint no-fallthrough: 0 */
+            switch (section) {
+                case 'allUpload':
+                    props.push('node');
+                case 'belowNode':
+                    props.push('selectedFile', 'file');
+                case 'belowFile':
+                    props.push('convertOrCopy');
+                case 'belowConvertOrCopy':
+                    props.push('nodeTitle');
+                    break;
```

Comment:
This is really the perfect use case for a [switch fall-through](http://eslint.org/docs/rules/no-fallthrough). Some say never use it, but in this case, it's the ideal solution.